### PR TITLE
Resolves #117: Support feeding in the datacenter_id option from start up command to …

### LINKED ISSUE #117: Supporting data center awareness
When running in a multi-datacenter FDB configuration, we encourage FDB clients to set the datacenter_id database option so that they know which instances are in their local DC and will prefer connecting to them. Can we add a flag in the document layer to set this option for its connection to FDB?

Reference on this database option: https://apple.github.io/foundationdb/configuration.html?highlight=datacenter_id#specifying-datacenters

### PR DESCRIPTION
**Pull request by dongxinEric**
…the FDB Database created.

### CODE REVIEW

**Comment by dongxinEric** [comment]
@brownleej expected syntax is the same for an optional argument: `--fdb_datacenter_id=SOME_AWESOME_ID`

**Comment by dongxinEric** [comment]
#117

File: src/DocLayer.actor.cpp [hunk 1]
```diff
@@ -875,6 +882,13 @@ int main(int argc, char** argv) {
 			}
 			break;
 		}
+		case OPT_FDB_DC_ID: {
+			const char* fdbDatacenterIdStr = args.OptionArg();
+			if (fdbDatacenterIdStr) {
+				fdbDatacenterid = Optional<StringRef>(LiteralStringRef(fdbDatacenterIdStr));
```
Comment (apkar) [inline src/DocLayer.actor.cpp]: `LiteralStringRef` is designed to use with constant strings `const char *`. It calls `sizeof()` on the string, that would be quite big. Best way to convert from `std::string` to `StringRef` is `StringRef(s)`. Also by passing `StringRef` we have to worry about destruction of string, which is fine here as it's in `main` which lives forever. I think here best way is to just pass `std:string` all the way. And convert to `StringRef` while calling `setOption`.

File: src/DocLayer.actor.cpp [hunk 2]
```diff
@@ -642,6 +648,7 @@ int main(int argc, char** argv) {
 	NetworkOptionsT client_network_options;
 	std::string metricReporterConfig;
 	char* metricPluginPath = nullptr;
+	Optional<StringRef> fdbDatacenterid;
```
Comment (apkar) [inline src/DocLayer.actor.cpp]: s/fdbDatacenterid/fdbDatacenterID

File: src/DocLayer.actor.cpp (hunk 1 above)
Comment (dongxinEric) [reply src/DocLayer.actor.cpp]: Sure thing.

Comment (apkar) [reply src/DocLayer.actor.cpp]: Not sure what I was reading. Ignore my comment `std::string`. But, please just pass around "const char*` instead `StringRef`

File: src/DocLayer.actor.cpp (hunk 2 above)
Comment (dongxinEric) [reply src/DocLayer.actor.cpp]: Done. 
67c9969

File: src/DocLayer.actor.cpp (hunk 1 above)
Comment (dongxinEric) [reply src/DocLayer.actor.cpp]: Done  67c9969

**Comment by dongxinEric** [comment]
@ajbeamon can you take a look at this? I briefly checked where the `datacenter_id` got used in the `fdbclient` code and it seems like OK to allow user to pass in random string, which in worse scenario will only make FDB thinks this client is not in any DC that it resides in and fall back to use latency.

File: src/DocLayer.actor.cpp [hunk 3]
```diff
@@ -596,6 +603,8 @@ void printHelp(const char* name) {
   --proxy-ports LISTEN_PORT MONGODB_PORT
              Runs Document Layer in proxy mode on LISTEN_PORT proxying all commands
              to MongoDB server running on MONGODB_PORT.
+  --fdb_datacenter_id DC_ID
+             Specify the datacenter id to use when connecting to the FDB that's run in multi-dc mode
```
Comment (ajbeamon) [inline src/DocLayer.actor.cpp]:
```suggestion
             The id of the preferred datacenter to use when connecting to a FoundationDB cluster 
             that's run in multi-dc mode
```

**Pull request merged**